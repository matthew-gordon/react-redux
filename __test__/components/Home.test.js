import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'

// import Foo from '../../src/js/components/Foo';
import Home from '../../src/js/components/home/Home';

// Snapshot for Home React Component
describe('>>>H O M E --- Snapshot',()=>{

  it('+++capturing Snapshot of Home', () => {
      const renderedValue =  renderer.create(<Home />).toJSON()
      expect(renderedValue).toMatchSnapshot();
  });

});

describe('>>>H O M E --- Shallow render REACT COMPONENTS', () => {

  it('+++ should render without throwing an error', () => {
    expect(shallow(<Home />).contains(<div className="container Home"><h1>Landing Page</h1></div>)).toBe(true);
  });

  it('+++ should mount in a full DOM', function() {
    expect(mount(<Home />).find('.Home').length).toBe(1);
  });

  it('+++ should render to static html', () => {
    expect(render(<Home />).text()).toEqual('Landing Page');
  });

});
