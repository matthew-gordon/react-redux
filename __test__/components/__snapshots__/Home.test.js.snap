// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>H O M E --- Snapshot +++capturing Snapshot of Home 1`] = `
<div
  className="container Home"
>
  <h1>
    Landing Page
  </h1>
</div>
`;
